### PR TITLE
Merge options if plugin was already added

### PR DESCRIPTION
**Pull request by pshevtsov**
Initially discussed in https://github.com/rstudio/dygraphs/pull/157#pullrequestreview-24231757

### CODE REVIEW

File: R/dependency.R [hunk 1]
```diff
@@ -56,6 +56,15 @@ dyPlugin <- function(dygraph, name, path, options = list(), version = "1.0") {
   if (length(options) == 0) {
     options <- JS("{}")
   }
+  # merge options if plugin was already added
```
Comment (jjallaire) [inline R/dependency.R]: I might be missing something, but doesn't this only pick up new options? (i.e. it wouldn't pick up overwriting of an existing option).

Comment (pshevtsov) [inline R/dependency.R]: No, it merges new options with the old ones:
```R
old.options <- list(color = "red",
                    fruit = "apple",
                    days = list("Monday", "Tuesday", "Wednesday"),
                    number = 42,
                    figure = "square")
options <- list(color = "blue",
                days = list("Saturday", "Sunday"),
                animal = "cat")

i <- match(names(old.options), names(options))
i <- is.na(i)
if (any(i)) {
  options[names(old.options)[which(i)]] = old.options[which(i)]
}

options
$color
[1] "blue"

$days
$days[[1]]
[1] "Saturday"

$days[[2]]
[1] "Sunday"


$animal
[1] "cat"

$fruit
[1] "apple"

$number
[1] 42

$figure
[1] "square"
```

Passing such `options` to a Dygraph plugin instance results in the following JavaScript object:
```json
{
   "animal" : "cat",
   "figure" : "square",
   "days" : [
      "Saturday",
      "Sunday"
   ],
   "fruit" : "apple",
   "number" : 42,
   "color" : "blue"
}
```

**Comment by jjallaire** [comment]
Could you add a NEWS item for this?

**Comment by pshevtsov** [comment]
@jjallaire Done https://github.com/rstudio/dygraphs/pull/159

**Pull request merged**